### PR TITLE
Milestone_3 and Milestone_4 tasks implementation

### PR DESCRIPTION
**Pull request by korsarking**
Completed 7 tasks from Milestone#3:
1.	Start a timer for my task (user send task id and receive successful response after logging the start of task in DB);
2.	Stop timer for the started task - user send task id and receive successful response after adding a time log for this task with duration of work for current date
3.	Add time log for a task on a specific date - user manually send in task id, date, duration in minutes and receive successful response of save
4.	Get a list of time logs records by task - user send task id and receive list of all time logs created for this task
5.	Change get list of tasks endpoint get receive the sum of the time in minutes for each task
6.	Get the logged time in last month - user send a request and receive total amount of time logged by him in last month
7.	Get top 20 tasks in last month by time - user send request and receive list of id, title, time amount of tasks with bigger logged amount of time

Completed 8 tasks from Milestone#4:
1.	Added unit tests for each endpoint;
2.	Added coverage package and generate report to be sure in 100% test coverage;
3.	Installed PostgreSQL docker container and implemented PostgreSQL database usage;
4.	Created a script that will add random 25.000 tasks and 50.000 time logs;
5.	Installed PGHero docker container and connect it to database;
6.	Manually tested all endpoints and checked with PGHero;
7.	Installed Redis docker container and configurated Django to cache with 1 minute TTL the endpoint with Top 20 tasks by time for each user;
8.	Created Dockerfile for application to run it with docker-compose command

### CODE REVIEW

File: .env.dev [hunk 1]
```diff
@@ -0,0 +1,10 @@
+SECRET_KEY=django-insecure-$(zi(3y@43nr)a*3t#+au-uzp)li($)ya8u*h)#uk8)$0o4x2$
+DEBUG=True
+EMAIL_BACKEND=django.core.mail.backends.smtp.EmailBackend
+EMAIL_URL=submission://korsarkings@gmail.com:hrfkyfoidpfpppof@smtp.gmail.com:587?tls=True&_default_from_email=korsarkings333@gmail.com
```
Comment (vladmotpan) [inline .env.dev]: Nu e bine sa ai secrets-uri pe git. Mai bine sa lasi EMAIL_URL gol sau cu un placeholder.

File: .env.dev [hunk 2]
```diff
@@ -0,0 +1,10 @@
+SECRET_KEY=django-insecure-$(zi(3y@43nr)a*3t#+au-uzp)li($)ya8u*h)#uk8)$0o4x2$
```
Comment (vladmotpan) [inline .env.dev]: SECRET_KEY mai bine sa fie placeholder

File: apps/common/views.py [hunk 3]
```diff
@@ -3,40 +3,15 @@
 from rest_framework.response import Response
 
 
-# Create your views here.
-
-
 class HealthView(GenericAPIView):
```
Comment (vladmotpan) [inline apps/common/views.py]: Aici e posibil fara metoda `get_queryset`. Ea nu are sens ca nu se foloseste. Poti sa folosessti `views.APIView`. In loc de `GenericAPIView`.

File: apps/tasks/models.py [hunk 4]
```diff
@@ -11,14 +16,60 @@ class Status(models.TextChoices):
...
+            duration = timezone.now() - self.started_at
+
+            self.is_started = False
+            self.duration = duration
+            self.started_at = timezone.localtime(self.started_at)
+            self.save(update_fields=["is_started", "started_at"])
+
+            TimeLog.objects.create(owner=self.owner, task=self.task, duration=duration, started_at=self.started_at)
+        else:
+            raise ValidationError({"detail": f"Task id:{self.id} has no ongoing timer."})
```
Comment (vladmotpan) [inline apps/tasks/models.py]: Validarea nu e bine sa fie in model method. Cel mai bine sa fie in serializer sau in view daca nu folosesti serailizer.

**Review by vladmotpan — CHANGES_REQUESTED** [review]

File: .env.dev (hunk 1 above)
Comment (korsarking) [reply .env.dev]: Done

File: .env.dev (hunk 2 above)
Comment (korsarking) [reply .env.dev]: Done

File: apps/common/views.py (hunk 3 above)
Comment (korsarking) [reply apps/common/views.py]: Done

File: apps/tasks/models.py (hunk 4 above)
Comment (korsarking) [reply apps/tasks/models.py]: Done

**Review by vladmotpan — APPROVED** [review]

**Pull request merged**